lib/Parse/ParseExpr.cpp

```diff
@@ -35,6 +35,10 @@
 
 using namespace swift;
 
+bool Token::isEditorPlaceholder() const {
+  return is(tok::identifier) && Identifier::isEditorPlaceholder(getRawText());
```

Comment:
Technically, `Identifier::isEditorPlaceholder()` don't do the job anymore? Is there a way to distinguish `` `<#placeholder#>` `` from `<#placeholder#>` after parsing? What `ValueDecl::shouldHideFromEditor()` should do?

Reply:
> Is there a way to distinguish `` `<#placeholder#>` `` from `<#placeholder#>` after parsing?

Unfortunately not, my preference here would be to ban placeholders from raw identifiers but that would probably need evolution discussion.

> What `ValueDecl::shouldHideFromEditor()` should do?

Oh interesting, I didn't know we did that for placeholders. In practice I don't expect people to ever actually use placeholders in raw identifiers, so I'm not sure this will be much of an issue in practice. But yeah, ideally we'd ban them.